diff --git a/dist/index.cjs b/dist/index.cjs
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -553,6 +553,11 @@
       );
       return utxos;
     },
+    ...(typeof api.getCollateral !== "function" ? {} : {
+      getCollateral: async () => (await api.getCollateral() || []).map(
+        (utxo) => (0, import_utils2.coreToUtxo)(CML.TransactionUnspentOutput.from_cbor_hex(utxo))
+      )
+    }),
     getUtxosCore: async () => {
       const utxos = config.overriddenUTxOs !== void 0 ? config.overriddenUTxOs.map(import_utils2.utxoToCore) : (await api.getUtxos() || []).map(
         (utxo) => CML.TransactionUnspentOutput.from_cbor_hex(utxo)
diff --git a/dist/index.js b/dist/index.js
--- a/dist/index.js
+++ b/dist/index.js
@@ -517,6 +517,11 @@
       );
       return utxos;
     },
+    ...(typeof api.getCollateral !== "function" ? {} : {
+      getCollateral: async () => (await api.getCollateral() || []).map(
+        (utxo) => coreToUtxo(CML.TransactionUnspentOutput.from_cbor_hex(utxo))
+      )
+    }),
     getUtxosCore: async () => {
       const utxos = config.overriddenUTxOs !== void 0 ? config.overriddenUTxOs.map(utxoToCore) : (await api.getUtxos() || []).map(
         (utxo) => CML.TransactionUnspentOutput.from_cbor_hex(utxo)
